openpype/hosts/maya/plugins/publish/validate_skeleton_rig_output_ids.py

```diff
@@ -0,0 +1,126 @@
+        return list(invalid_matches.keys())
+
+    @classmethod
+    def get_invalid_matches(cls, instance, compute=False):
+        invalid = {}
+
+        if compute:
+            skeletonMesh_set = instance.data["rig_sets"].get(
+                "skeletonMesh_SET")
+            if not skeletonMesh_set:
+                instance.data["mismatched_output_ids"] = invalid
+                return invalid
+
+            instance_nodes = cmds.sets(
+                skeletonMesh_set, query=True, nodesOnly=True)
+
+            instance_nodes = cmds.ls(instance_nodes, long=True)
+            if not instance_nodes:
+                return {}
+            for node in instance_nodes:
+                shapes = cmds.listRelatives(node, shapes=True, fullPath=True)
+                if shapes:
+                    instance_nodes.extend(shapes)
+
+            scene_nodes = cmds.ls(type="transform", long=True)
+            scene_nodes += cmds.ls(type="mesh", long=True)
+            scene_nodes = set(scene_nodes) - set(instance_nodes)
+
+            scene_nodes_by_basename = defaultdict(list)
+            for node in scene_nodes:
+                basename = get_basename(node)
+                scene_nodes_by_basename[basename].append(node)
+
+            for instance_node in instance_nodes:
+                basename = get_basename(instance_node)
+                if basename not in scene_nodes_by_basename:
+                    continue
+
+                matches = scene_nodes_by_basename[basename]
+
+                ids = set(get_id(node) for node in matches)
+                ids.add(get_id(instance_node))
+
+                if len(ids) > 1:
+                    cls.log.error(
+                        "\"{}\" id mismatch to: {}".format(
+                            instance_node, matches
+                        )
+                    )
+                    invalid[instance_node] = matches
+
+            instance.data["mismatched_output_ids"] = invalid
+        else:
+            invalid = instance.data["mismatched_output_ids"]
+
+        return invalid
+
+    @classmethod
+    def repair(cls, instance):
+        invalid_matches = cls.get_invalid_matches(instance)
+
+        multiple_ids_match = []
+        for instance_node, matches in invalid_matches.items():
+            ids = set(get_id(node) for node in matches)
+
+            # If there are multiple scene ids matched, and error needs to be
+            # raised for manual correction.
+            if len(ids) > 1:
+                multiple_ids_match.append({"node": instance_node,
+                                           "matches": matches})
+                continue
+
+            id_to_set = next(iter(ids))
+            set_id(instance_node, id_to_set, overwrite=True)
+
+        if multiple_ids_match:
+            raise PublishValidationError(
+                "Multiple matched ids found. Please repair manually: "
+                "{}".format(multiple_ids_match)
+            )
```

Comment:
Are we sure we can raise PublishValidationError reports from a `repair` action. I'm pretty sure the publisher does not support this. @iLLiCiTiT do you know?